regolith/builders/coabuilder.py

```diff
@@ -3,40 +3,211 @@
 
+
 def mdy_date(month, day, year, **kwargs):
+    """Make a date object."""
     if isinstance(month, str):
         month = month_to_int(month)
     return dt.date(year, month, day)
 
 
 def mdy(month, day, year, **kwargs):
+    """Format the date to a string mm/dd/yy."""
     return "{}/{}/{}".format(
         str(month_to_int(month)).zfill(2), str(day).zfill(2), str(year)[-2:]
     )
 
 
+def fitler_since_date(pubs, since_date):
+    """Filter the publications after the since_date."""
+    for pub in pubs:
+        if is_since(pub.get("year"), since_date.year,
+                    pub.get("month", 1), since_date.month):
+            if not pub.get("month"):
+                print("WARNING: {} is missing month".format(
+                    pub["_id"]))
+            if pub.get("month") == "tbd".casefold():
+                print("WARNING: month in {} is tbd".format(
+                    pub["_id"]))
+            yield pub
+
+
+def get_since_date(rc):
+    """Get the since_date from command."""
+    if isinstance(rc, str):
+        since_date = dt.datetime.strptime(rc, '%Y-%m-%d').date() - relativedelta(months=NUM_MONTHS)
+    else:
+        since_date = dt.date.today() - relativedelta(months=NUM_MONTHS)
+    return since_date
+
+
+def get_coauthos_from_pubs(pubs):
+    """Get co-authors' names from the publication."""
+    my_collabs = []
+    for pub in pubs:
+        my_collabs.extend(
+            [
+                collabs for collabs in
+                (names for names in pub.get('author', []))
+            ]
+        )
+    return my_collabs
+
+
+def query_people_and_instituions(rc, names):
+    """Get the people and institutions names."""
+    people, institutions = [], []
+    for person_name in names:
+        person_found = fuzzy_retrieval(all_docs_from_collection(
+            rc.client, "people"),
+            ["name", "aka", "_id"],
+            person_name)
+        if not person_found:
+            person_found = fuzzy_retrieval(all_docs_from_collection(
+                rc.client, "contacts"),
+                ["name", "aka", "_id"], person_name)
+            if not person_found:
+                print(
+                    "WARNING: {} not found in contacts. Check aka".format(
+                        person_name))
+            else:
+                people.append(person_found['name'])
+                inst = fuzzy_retrieval(all_docs_from_collection(
+                    rc.client, "institutions"),
+                    ["name", "aka", "_id"],
+                    person_found["institution"])
+                if inst:
+                    institutions.append(inst["name"])
+                else:
+                    institutions.append(person_found.get("institution", "missing"))
+                    print("WARNING: {} missing from institutions".format(
+                        person_found["institution"]))
+        else:
+            people.append(person_found['name'])
+            pinst = person_found.get("employment",
```

Comment:
I did this but I don't like it.  Let's add a # fixme line here.  Right now it just looks at the first employment item to get "current" employment, but this is very brittle.  Better, would be to find the most recent employment.